pkg/kube/translation/translator.go

```diff
@@ -388,3 +398,27 @@ func (t *translator) TranslateIngress(ing kube.Ingress, args ...bool) (*Translat
 		return nil, fmt.Errorf("translator: source group version not supported: %s", ing.GroupVersion())
 	}
 }
+
+func (t *translator) TranslateOldRoute(ar kube.ApisixRoute) (*TranslateContext, error) {
+	switch ar.GroupVersion() {
+	case config.ApisixV2:
+		return t.translateOldRouteV2(ar.V2())
+	case config.ApisixV2beta3:
+		return t.translateOldRouteV2beta3(ar.V2beta3())
+	case config.ApisixV2beta2:
+		return DefaultEmptyTranslateContext(), nil
+	default:
+		return nil, fmt.Errorf("translator: source group version not supported: %s", ar.GroupVersion())
+	}
+}
+
+func (t *translator) TranslateOldIngress(ing kube.Ingress) (*TranslateContext, error) {
+	switch ing.GroupVersion() {
+	case kube.IngressV1:
+		return t.translateOldIngressV1(ing.V1())
+	case kube.IngressV1beta1:
+		return t.translateOldIngressV1beta1(ing.V1beta1())
+	default:
+		return nil, fmt.Errorf("translator: source group version not supported: %s", ing.GroupVersion())
+	}
```

Comment:
Currently we have not removed support for IngressExtensionsV1beta1, still need to deal with it